signxml/processor.py

```diff
@@ -36,10 +36,18 @@ def parser(self):
             return self._default_parser
         return self._parser
 
-    def _fromstring(self, xml_string, **kwargs):
-        xml_node = etree.fromstring(xml_string, parser=self.parser, **kwargs)
-        for entity in xml_node.iter(etree.Entity):
-            raise InvalidInput("Entities are not supported in XML input")
+    def _fromstring(self, data, **kwargs):
+        try:
+            xml_node = etree.fromstring(data, parser=self.parser, **kwargs)
+            for entity in xml_node.iter(etree.Entity):
+                raise InvalidInput("Entities are not supported in XML input")
+        except etree.XMLSyntaxError as e:
```

Comment:
This is not safe. In general, we should not ignore XML syntax errors, should not try to parse arbitrary binary data as XML, and should not return from an except block.

Any configuration that references a binary data blob should be direct and explicit, not implicit.

Reply:
I get it. But I'm struggling to find a way to mark the data as a particular content type, so it would be treated properly in various places of the code. 

So what would you think about adding a `non_xml` boolean parameter to `sign` method and propagating it through `_unpack`, `get_root`, etc. methods to `_fromstring` invocations? I don't see a clear way to attach that info to the data variable.

Reply:
OK, I'll think about the best way to organize this more over the weekend and come back with a recommendation here.